spec/bowl_spec.rb

```diff
@@ -0,0 +1,255 @@
+require 'bowl'
+
+describe BowlingScorer do
+  let(:bowl) {BowlingScorer.new}
+
+  before(:each) do
+    bowl.reset_scores_and_frames
+  end
+
+  def simulate_a_game_until_last_shot
+    bowl.add_frame(5,5)
+    bowl.add_frame(3,5)
+    bowl.add_frame(7,0)
+    bowl.add_frame(2,5)
+    bowl.add_frame(10,0)
+    bowl.add_frame(3,5)
+    bowl.add_frame(7,0)
+    bowl.add_frame(2,5)
+    bowl.add_frame(7,2)
+    ## By the end of round 9, player has 84 points scored.
+    ## All subsequent endgame tests are based on this one
+  end
+
+  context "add_frame method" do  
+    it "fails if user tries to knock more than 10 in one frame" do
+      expect{bowl.add_frame(6,5)}.to raise_error "Looks like you hit the next lane or something, cant knock more than 10"
+    end
+
+    it "fails if user inputs negative numbers" do
+      expect{bowl.add_frame(-1,5)}.to raise_error "Smells like invalid input"
+    end
+
+    it "fails if user inputs negative numbers" do
+      expect{bowl.add_frame(4,-5)}.to raise_error "Smells like invalid input"
+    end
+
```

Comment:
Good fail scenarios but would suggest one for non numeric input, floating point and maybe other types but that's just me nitpicking.